### ISSUE TITLE
Test proxy doesn't normalize line endings

### ISSUE THREAD

**Issue by pakrym** [Azure/azure-sdk-tools#2453]
![image](https://user-images.githubusercontent.com/1697911/146276401-deea4098-ac13-4ee4-8910-5a203359dd80.png)

@JoshLove-msft @scbedd

**Comment by scbedd** [comment]
Doing some digging into this. Found [this issue](https://github.com/dotnet/runtime/issues/54715) which is basically exactly our scenario.

So I think we need to create a custom `Utf8JsonWriter` and override the `WriteIndent` call to insider `\n` instead of `Environment.NewLine`.

We should also update our we _load_ the recording to ignore `crlf` when deserializing from the recording file.

**Comment by JoshLove-msft** [comment]
> Doing some digging into this. Found [this issue](https://github.com/dotnet/runtime/issues/54715) which is basically exactly our scenario.
> 
> So I think we need to create a custom `Utf8JsonWriter` and override the `WriteIndent` call to insider `\n` instead of `Environment.NewLine`.
> 
> We should also update our we _load_ the recording to ignore `crlf` when deserializing from the recording file.

I think the issue is with the actual content, not the formatted JSON. When matching we renormalize the JSON anyways so it should handle this for us. But the issue happens when someone adds line breaks based on Environment.NewLine from their own test.

**Comment by scbedd** [comment]
> When matching we renormalize the JSON anyways so it should handle this for us.

Yeah I'm seeing this now.

> But the issue happens when someone adds line breaks based on Environment.NewLine from their own test into the request content

Hmmm. So add a hardcoded replace on `\r\n` if the request content is a `text` content type? 

> when someone adds line breaks based on Environment.NewLine from their own test into the request content

I can see how this might be the proxy's place to do this, but I'm super leery towards it. Bright side is that this is _not_  a breaking change. Existing recordings would just have `crlf` content silently replaced when matching.

**Comment by JoshLove-msft** [comment]
Agreed that this shouldn't be a breaking change. I think essentially normalizing the line breaks when matching is the way to go.
